bsip-0071.md

```diff
@@ -0,0 +1,93 @@
+
+# Rationale
+
+We now have several choices on handling bad debt:
+
+1. Global Settlement
+2. Global Settlement Protection via Price Feeding
+3. Convert Bad Debt Positions to Limit Orders
+4. Take Over only Under-Collateralized Debt Positions without GS (aka Partial GS)
+5. Prevent Global Settlement
+
+We need to do a deep review on bad debt before evaluating above solutions.
+
+Bad debt means in some debt positions the debt cannot be fully paid by selling the collaterals via margin call/force settlement at the market price.  In other words, the relevant smartcoin loses sufficient collateral to back the value and will possibly devalue.
+
+At this moment, the system should accept the fact that the smartcoin will possibly devalue, find ways to minimize the impact to different parties in the market and the time to revive.
+
+In my view, a good way to handle bad debt need to follow below principles:
+
+1.  Do not punish good traders who have managed to maintain their collateral ratio well.
+2.  Do not stop the smartcoin features, including borrowing, margin call and force settlement, and all these operations should refer to the same price.
+3.  Ensure the debt positions/margin call orders be settled in the order from lower CR to higher CR.
+
+In above mentioned 5 solutions, only "5. Prevent Global Settlement" fulfills all 3 principles.
+
+Partial GS is another attractive solution: it is similar to Global Settlement but differs in that (a) only takes over the bad debt positions and moves them to a settlement pool without touching the debt positions with CR>1, and (b) users can issue force settlement from the pool, from the margin call orders, or from the good debt position depending on which has the lowest CR.
+
+The problem of Partial GS is that borrowing and force settlement may refer to different prices which will lead to obvious confusion and unfairness.
+
+The "Prevent Global Settlement" solution adopts a new idea to handle bad debt: while bad debt appears, the smartcoin will be devaluated at a ratio of the lowest CR from among the bad debt positions, yet all the smartcoin trading features, including borrowing, margin call and force settlement, will all switch to refer to GS price to ensure the continuity and fairness of all the features.
+
+# Specifications
+
+Add one flag "Prevent Global Settlement" to each smartcoin asset.
+
+Introduce a new parameter: settlement price.  Its value shall be calculated as:
+
+```
+if flag "Prevent Global Settlement" is enabled
+	setlement price =  max(FP_M, P_gs)
+else
+	settlement price =  FP_M
+```
+
+where FP<sub>M</sub> is the median of the prices published by witnesses; witnesses should always feed the real market price.
+
+All calculations of debt position collateral ratios and trading price determinations in borrowing, margin call and force settlement shall refer to the settlement price instead of directly to the feed price.
```

Comment:
Yes, this is what we want to make clear in the BSIP.

Reply:
Does the following text capture the desired intention?

---

The calculations of debt position collateral ratios and trading price determinations for margin calls and force settlement shall refer to the settlement price instead of directly to the feed price.

The calculations of debt position collateral ratios _when creating a new_ borrowing position shall refer to the feed price instead of the settlement price.

Reply:
Hm. I'm unsure what the desired intention is. :-/
I think we agree that the relevant price when updating call_orders is MCR*feed_price, which is the same as the price that determines if a given call_order is margin called.
I think we also agree that the execution price for forced settlements (aka settlement_price) is min(feed_price,CR of least collateralized call_order) and the execution price for call_orders with CR < feed_price*MSSR is min(feed_price*MSSR, CR), simply because these call_orders cannot pay more than that.
This leaves us with the execution price for call_orders with feed_price*MCR > CR > feed_price*MSSR. IMO these should be margin called at feed_price*MSSR, as usual. The rule then boils down to "pay feed_price*MSSR if you can, otherwise as much as you can".
The downside of this simple rule is that margin calls with a better CR execute at a better price, which means that those with worse CR are less likely to be bought up, which is counterproductive. OTOH, having all margin calls execute at the price of the least collateralized one could lead to all margin calls drifting away from the market price with no chance of getting filled.

Reply:
One option is, when a big limit order buying into multiple margin calls, fill them at the same price; after done, recalculate settlement_price, that said, if the asset was GS-protected, after some call orders being bought, the settlement_price should decrease; then, the next limit order buying into margin calls would buy at better price ("better" for the new limit order in comparison the the previous limit order).

Another option is, when a big limit order buying into multiple margin calls, after each margin call got filled, recalculate settlement_price, this means the next call order will be bought at lower price (in favor of the buyer). I think this approach is better.

Reply:
What's more important, having the least collateralized call_order filled, or have as much margin-called debt filled as possible? Neither option is perfect.

Reply:
Ok, then margin calls should "pay feed_price*MSSR if they can, otherwise pay as much as they can".

Reply:
What is your opinion @abitmore ?

Reply:
> margin calls should "pay feed_price*MSSR if they can, otherwise pay as much as they can".

I agree with this.

Reply:
Suggested wording:

> This `settlement_price` is the price that is used when executing forced settlements. Note that when a large forced settlement fills up a debt position completely, P_gs is likely to change. The resulting new `settlement_price` will be used when settling against the next debt position.
`FP_M` is still used for determining if a debt position is margin called, as before, at `FP_M*MCR`.
When matching margin calls against limit orders or settlement requests, the execution price is `min(FP_m*MSSR,collateral/debt)`. I. e. margin positions with a CR below MSSR pay with their own CR in an attempt to recover the full debt with the full collateral.

Perhaps also redefine `settlement_price` above to include the settlement_offset.

Reply:
@pmconrad Please check if I added the settlement offset correctly

Reply:
Ah, no. MSSR is not the settlement offset. I don't think we have a going acronym for it.
The correct factor would be `(1 - settlement_offset)`. (I. e. if offset is 5% then the settler receives 0.95 times as much collateral as FP_M alone would indicate.)

Reply:
Thanks let's try again

Reply:
Thanks. Please insert paragraphs like in my comment above, for better readability. (Before "FP_M is still used" and before "When matching margin calls".) Apparently, the `.md` parser handles paragraphs differently than the comment function here.

Reply:
After reconsidering, I think the settlement price should be defined as 
`max(FP_M * (1 - settlement_offset), P_gs)`. I don't see why an undercollateralized position should be rewarded by the settlement_offset. Also, the reason why the settlement_offset was increased (premium) doesn't apply here.

Reply:
I think the discussion about `settlement_offset` is in scope of #181 (BSIP73).